### PR TITLE
feat: Add weak event registration generation

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	private record GenerationCandidateData
+	{
+		public GenerationCandidateData(ISymbol symbol, AttributeData attribute)
+		{
+			var eventSymbol = symbol as IEventSymbol ?? throw new InvalidOperationException("Symbol must be an event");
+
+			var isDependencyObject = eventSymbol.ContainingType.AllInterfaces
+				.Any(t => t.ToDisplayString(_fullyQualifiedWithoutGlobal) == XamlConstants.Types.DependencyObject);
+
+			ContainingTypeIsCandidate = eventSymbol.ContainingType.TypeKind == TypeKind.Class && isDependencyObject;
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs:32]: Any reason we limit generation to DependencyObject?

File: src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+			ContainingTypeFullyQualifiedName = eventSymbol.ContainingType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+			ContainingTypeHintName = HashBuilder.BuildIDFromSymbol(eventSymbol.ContainingType);
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs:34]: I guess we might not need a hash. Using `eventSymbol.ContainingType.GetFullMetadataNameForFileName()` should likely be sufficient.

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/RadioButtons/RadioButtons.cs [hunk 3]
```diff
@@ -72,11 +77,6 @@ ItemsRepeater GetRepeater()
 				{
 					repeater.ItemTemplate = m_radioButtonsElementFactory;
 
-					repeater.ElementPrepared += OnRepeaterElementPrepared;
-					repeater.ElementClearing += OnRepeaterElementClearing;
-					repeater.ElementIndexChanged += OnRepeaterElementIndexChanged;
-					repeater.Loaded += OnRepeaterLoaded;
-					repeater.Unloaded += OnRepeaterUnloaded;
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/RadioButtons/RadioButtons.cs:79]: Is this intentionally removed?

File: src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs:32]: At this time, there isn't any in code, but the assumption is that we may need to use loaded/unloaded specifics if performance becomes a concern with weak references. If we need to expand this for other reasons, we'll adjust.

File: src/SourceGenerators/Uno.UI.SourceGenerators.Tests/WeakEventGeneratorTests/Given_WeakEventGenerator.cs
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators.Tests/WeakEventGeneratorTests/Given_WeakEventGenerator.cs:1]: We may need to disable in UWP similar to

https://github.com/unoplatform/uno/blob/a0bb1df67c906e4dea94c99776b7f82334a6210b/src/SourceGenerators/Uno.UI.SourceGenerators.Tests/Uno.UI.SourceGenerators.Tests.csproj#L293

**Comment by jeromelaban** [comment]
Possible memory leaks:

image

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/FrameworkElement.skia.cs [hunk 4]
```diff
@@ -76,6 +78,8 @@ public event TypedEventHandler<FrameworkElement, object> Loading
 		}
 
 		private event RoutedEventHandler _loaded;
+
+		[global::Uno.UI.Xaml.GeneratedWeakEvent]
 		public event RoutedEventHandler Loaded
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/FrameworkElement.skia.cs:83]: I think WinUI's Loaded event isn't weak. So not really sure if we should do it that way

**Pull request closed without merging**